cypress/integration/02 - items.ts

```diff
@@ -10,125 +10,78 @@ const itemSearch = {
-            .find("pl-items-list")
-            .find("main pl-virtual-list pl-scroller")
-            .find("div.content")
-            .children("div")
-            .should("have.length", 1);
-
-        // Confirm we find the right one
-        cy.get("pl-app")
-            .find("pl-items")
-            .find("pl-items-list")
-            .find("main pl-virtual-list pl-scroller")
-            .find("div.content pl-vault-item-list-item")
-            .find("div > div > div.semibold")
-            .should("include.text", testItem.name);
-
-        // Click clear search sign
-        cy.get("pl-app")
-            .find("pl-items")
-            .find("pl-items-list")
-            .find("pl-input#filterInput")
-            .find("pl-button.slim")
-            .click();
-
-        // Click search sign
-        cy.get("pl-app").find("pl-items").find("pl-items-list").find("pl-button:eq(3)").click();
-
-        // Find non-existent Item
-        cy.get("pl-app")
-            .find("pl-items")
-            .find("pl-items-list")
-            .find("pl-input#filterInput")
-            .find("input")
-            .type(itemSearch.nonexistent, { force: true });
-
-        // Confirm we find none
-        cy.get("pl-app")
-            .find("pl-items")
-            .find("pl-items-list")
-            .find("main pl-virtual-list pl-scroller")
-            .find("div.content")
-            .children("div")
-            .should("have.length", 0);
-
-        cy.get("pl-app")
-            .find("pl-items")
-            .find("pl-items-list")
-            .find("main > div.centering")
-            .should("contain.text", "did not match any items");
+        cy.unlock(email);
+
+        cy.doWithin(["pl-app", "pl-items", "pl-items-list"], () => {
+            // Click search sign
+            cy.get("pl-button:eq(3)").click();
+
+            // Find Item
+            cy.typeWithin("pl-input#filterInput", itemSearch.existing, { force: true });
+
+            // Confirm we only find one
+            cy.get("main pl-virtual-list pl-scroller div.content").children("div").should("have.length", 1);
+
+            // Confirm we find the right one
+            cy.get("main pl-virtual-list pl-scroller")
+                .find("div.content pl-vault-item-list-item")
+                .find("div > div > div.semibold")
+                .should("include.text", testItem.name);
+
+            // Click clear search sign
+            cy.get("pl-input#filterInput pl-button.slim").click();
+
+            // Click search sign
+            cy.get("pl-button:eq(3)").click();
+
+            // Find non-existent Item
+            cy.typeWithin("pl-input#filterInput", itemSearch.nonexistent, { force: true });
+
+            // Confirm we find none
+            cy.get("main pl-virtual-list pl-scroller div.content").children("div").should("have.length", 0);
+
+            cy.get("main > div.centering").should("contain.text", "did not match any items");
+        });
```

Comment:
I think this is a great example of how the `doWithin` command makes things more readable.